webui/src/components/editrecord.jsx

```diff
@@ -666,21 +783,32 @@ const EditRecord = React.createClass({
                         { this.props.isDraft ? this.renderFileBlock() : false }
                     </div>
                     <div className="col-xs-12">
+                    <fieldset disabled={this.state.readOnly}>
                         <form className="form-horizontal" onSubmit={this.updateRecord}>
                             { this.renderFieldBlock(null, rootSchema) }
 
                             { !blockSchemas ? false :
                                 blockSchemas.map(([id, blockSchema]) =>
                                     this.renderFieldBlock(id, (blockSchema||Map()).get('json_schema'))) }
                         </form>
+                    </fieldset>
                     </div>
                 </div>
                 <div className="row">
-                    <div className="form-group submit row" style={{margin:'2em 0', paddingTop:'2em', borderTop:'1px solid #eee'}}>
-                        {pairs(this.state.errors).map( ([id, msg]) =>
-                            <div className="col-sm-offset-3 col-sm-9 alert alert-warning" key={id}>{msg} </div>) }
-                        { this.props.isDraft ? this.renderSubmitDraftForm() : this.renderUpdateRecordForm() }
-                    </div>
+                    {this.state.record.get('publication_state') == 'submitted' && this.state.readOnly ?
+                        <div className="form-group submit row" style={{margin:'2em 0', paddingTop:'2em', borderTop:'1px solid #eee'}}>
+                            <div className="col-sm-offset-3 col-sm-9">
+                                <p> Note that by editing the record, it will be revoked and won't being reviewd by your community admin anymore. You will need to submit it again. </p>
+                                <button type="submit" className={"btn btn-default btn-block btn-primary btn-danger "} onClick={this.editSubmittedRecord}>Edit</button>
+                            </div>
+                        </div>
+                    :
```

Comment:
can we make this better?  move to separate functions?

Reply:
I'm not sure what you mean. You mean moving the condition part to a separate function?

Reply:
I was thinking of moving the divs into separate functions, but that may make the code even more difficult to follow. Let's leave it for the moment.